t/endpoint.t

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env perl
+
+use 5.028;
+use warnings;
+use experimental qw(signatures);
+
+use Test::Lib;
+use Test::PromAlertProxy;
+use Test::More;
+use Test::Time time => Test::PromAlertProxy->now;
+
+use IO::Async::Loop;
+
+use Plack::Test;
+use HTTP::Request::Common;
+use JSON::MaybeXS;
+
+use PromAlertProxy::Hub;
+
+use PromAlertProxy::Logger '$Logger' => { init => {
+  ident    => 'promalertproxy',
+  to_self  => 1,
+} };
+
+my $loop = IO::Async::Loop->new;
+
+my $hub = PromAlertProxy::Hub->new(loop => $loop);
```

Comment:
I wonder if, rather than having to pass in a loop object everywhere, you should just have the loop attr in Hub.pm just default to `IO::Async::Loop->new`, which would save you some typing in a bunch of places and you could still get at it later with $hub->loop.

Reply:
Mmm. Again, the idea is that it slots into the surrounding environment. But that doesn't prevent it, and a hypothetical future could still do it because its just an attribute. Ok, nice!

Reply:
ALSO: unless you're doing something _super weird_, IO::Async::Loop->new almost always does what you want, and tries very hard to find the One True Loop. From its docs:

> The constructed object is cached, and will be returned again by a subsequent call. The cache will also be set by a constructor on a specific subclass. This behaviour makes it possible to simply use the normal constructor in a module that wishes to interact with the main program's Loop, such as an integration module for another event system.

But also yes, as you say, you can also pass it in if you feel really strongly, even if ->new would get you the same thing.